server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreShardAllocationIT.java

```diff
@@ -0,0 +1,463 @@
+
+        logger.info(" --> add remote and non-remote nodes");
+        addRemote = false;
+        String nonRemoteNodeName = internalCluster().startNode();
+        addRemote = true;
+        String remoteNodeName = internalCluster().startNode();
+        internalCluster().validateClusterFormed();
+        DiscoveryNode nonRemoteNode = assertNodeInCluster(nonRemoteNodeName);
+        DiscoveryNode remoteNode = assertNodeInCluster(remoteNodeName);
+        assertFalse(nonRemoteNode.isRemoteStoreNode());
+        assertTrue(remoteNode.isRemoteStoreNode());
+
+        logger.info(" --> allocate primary shard on non-remote node");
+        client.admin()
+            .indices()
+            .prepareCreate(TEST_INDEX)
+            .setSettings(
+                Settings.builder()
+                    .put("index.number_of_shards", 1)
+                    .put("index.number_of_replicas", 1)
+                    .put("index.routing.allocation.include._name", nonRemoteNodeName)
+                    .put("index.routing.allocation.exclude._name", cmNodeName + "," + remoteNodeName)
+            )
+            .setWaitForActiveShards(ActiveShardCount.ONE)
+            .execute()
+            .actionGet();
+
+        logger.info(" --> verify allocation of primary shard");
+        RoutingTable routingTable = client.admin().cluster().prepareState().execute().actionGet().getState().getRoutingTable();
+        ShardRouting primaryShardRouting = routingTable.index(TEST_INDEX).shard(0).primaryShard();
+        assertTrue(primaryShardRouting.active());
+        assertNotNull(primaryShardRouting.currentNodeId());
+        assertEquals(primaryShardRouting.currentNodeId(), nonRemoteNode.getId());
+
+        logger.info(" --> verify non-allocation of replica shard");
+        ShardRouting replicaShardRouting = routingTable.index(TEST_INDEX).shard(0).replicaShards().get(0);
+        assertFalse(replicaShardRouting.active());
+        assertNull(replicaShardRouting.currentNodeId());
+        assertEquals(replicaShardRouting.state(), ShardRoutingState.UNASSIGNED);
+
+        logger.info(" --> set remote_store direction");
+        setDirection("remote_store");
+
+        logger.info(" --> allocate replica shard on remote node");
+        client.admin()
+            .indices()
+            .prepareUpdateSettings(TEST_INDEX)
+            .setSettings(
+                Settings.builder()
+                    .put("index.routing.allocation.include._name", remoteNodeName)
+                    .put("index.routing.allocation.exclude._name", cmNodeName + "," + nonRemoteNodeName)
+            )
+            .execute()
+            .actionGet();
+
+        logger.info(" --> verify non-allocation of replica shard");
+        routingTable = client.admin().cluster().prepareState().execute().actionGet().getState().getRoutingTable();
+        replicaShardRouting = routingTable.index(TEST_INDEX).shard(0).replicaShards().get(0);
+        assertFalse(replicaShardRouting.active());
```

Comment:
Same as above. The assertion should be on the reason for UNASSIGNED shard and the cluster state should be yellow.